ledger/src/sigverify_shreds.rs

```diff
@@ -267,7 +273,9 @@ pub fn verify_shreds_gpu(
 
 fn sign_shred_cpu(keypair: &Keypair, packet: &mut Packet) {
     let sig = shred::layout::get_signature_range();
-    let msg = shred::layout::get_signed_message_range(packet);
+    let msg = shred::layout::get_shred(packet)
+        .and_then(shred::layout::get_signed_message_range)
+        .unwrap();
     assert!(
```

Comment:
Is this assert superfluous after the `unwrap()` above?

Reply:
The assert is on `sig.end`, the unwrap is on `msg`.
Here it does not make any assumptions about the underlying layout of shreds.